spec/models/contest_problem_spec.rb

```diff
@@ -27,4 +27,14 @@
   def create_submit(participation, problem, stat, created_at)
     create(:submit, contest_problem: problem, author: participation.user, status: stat, created_at: created_at)
   end
+
+  describe '.hard_overdue?' do
+    context 'before deadline' do
+      it { expect(ContestProblem.hard_overdue?(create(:contest_problem, hard_deadline: Date.tomorrow).id)).to be false }
+    end
+
+    context 'after deadline' do
+      it { expect(ContestProblem.hard_overdue?(create(:contest_problem, hard_deadline: Date.yesterday).id)).to be true }
```

Comment:
Można zastosować  [syntax sugar do predicate matchers](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/predicate-matchers)
Przykład:
`it { expect(ContestProblem).to be_hard_overdue(create(:contest_problem, hard_deadline: Date.yesterday).id) }`
Kwestia co uważamy za czytelniejsze

Reply:
Miałem wprowadzić zmianę, ale wstrzymam się z powodu outputu, który zobaczyłem po uruchomieniu testu:
```
  .hard_overdue?
    before deadline
      should not be hard overdue 2824
    after deadline
      should be hard overdue 2825
```
Gdzie `2824` i `2825` to identyfikatory wyprodukowane w ramach testu przez FactoryGirl.